### PR TITLE
layers: Fix naming and type selection of external handles

### PR DESCRIPTION
**Pull request by artem-lunarg**
Use correct type (Flags vs FlagBits).
Use correct singular or plural form for handle type variables.

### CODE REVIEW

File: layers/state_tracker/device_memory_state.cpp [hunk 1]
```diff
@@ -19,12 +19,16 @@
 #include "state_tracker/device_memory_state.h"
 #include "state_tracker/image_state.h"
 
-static VkExternalMemoryHandleTypeFlags GetExportHandleType(const VkMemoryAllocateInfo *p_alloc_info) {
+// It is allowed to export memory into the handles of different types,
+// that's why we use set of flags (VkExternalMemoryHandleTypeFlags)
+static VkExternalMemoryHandleTypeFlags GetExportHandleTypes(const VkMemoryAllocateInfo *p_alloc_info) {
```
Comment (artem-lunarg) [inline layers/state_tracker/device_memory_state.cpp:24]: These two functions are the main motivation for this PR. The other changes are mostly mechanical rename.

**Review by spencer-lunarg — APPROVED** [review]
:100: like these long-needed cleanups!

**Comment by artem-lunarg** [comment]
Probably far from complete clean up, I did not try to find everything, only the parts that directly interface with my current VUID task.

**Review by arno-lunarg — APPROVED** [review]

**Comment by artem-lunarg** [comment]
Removed the build from internal CI for the last change, which is a comment update. Github CI should be enough for this.

**Pull request merged**